g2p_programs/models/entitlement.py

```diff
@@ -164,38 +168,10 @@ def approve_entitlement(self):
         sw = 0
         for rec in self:
             if rec.state in ("draft", "pending_validation"):
-                fund_balance = self.check_fund_balance(rec.cycle_id.program_id.id) - amt
-                if fund_balance >= rec.initial_amount:
-                    amt += rec.initial_amount
-                    # Prepare journal entry (account.move) via account.payment
-                    payment = {
-                        "partner_id": rec.partner_id.id,
-                        "payment_type": "outbound",
-                        "amount": rec.initial_amount,
-                        "currency_id": rec.journal_id.currency_id.id,
-                        "journal_id": rec.journal_id.id,
-                        "partner_type": "supplier",
-                    }
-                    new_payment = self.env["account.payment"].create(payment)
-                    rec.update(
-                        {
-                            "disbursement_id": new_payment.id,
-                            "state": "approved",
-                            "date_approved": fields.Date.today(),
-                        }
-                    )
-                else:
-                    raise UserError(
-                        _(
-                            "The fund for the program: %(program)s [%(fund).2f] "
-                            + "is insufficient for the entitlement: %(entitlement)s"
-                        )
-                        % {
-                            "program": rec.cycle_id.program_id.name,
-                            "fund": fund_balance,
-                            "entitlement": rec.code,
-                        }
-                    )
+                if rec.is_cash_entitlement:  # For Cash Entitlements
+                    amt = self._process_cash_base_entitlement(rec, amt)
+                else:  # For Non-cash Entitlements
+                    self._process_noncash_base_entitlement(rec)
```

Comment:
Should we rename `_process_cash_base_entitlement` to `_process_approve_entitlement`?

Reply:
Yes but I named the function like this to identify the cash and noncash entitlement.

Reply:
is that needed? as a manager will likely not implement the 2 and actually this function would not support it

Reply:
Done